aten/src/ATen/cuda/CUDABlas.cpp

```diff
@@ -1453,7 +1457,12 @@ void scaled_gemm(
   // rowwise isn't supported using cublaslt or older hipblaslt
   TORCH_INTERNAL_ASSERT(use_rowwise == false, "rowwise scaled_gemm not supported with blaslt");
 #endif
-  {
+  if (mat1_scale_dtype == kFloat8_e8m0fnu && mat2_scale_dtype == kFloat8_e8m0fnu) {
+    // scales are swapped in the cuBLAS kernel, swap them here to hide this from the user
```

Comment:
TBH I dont want to land this before we can figure out why this happening. This feels just wrong

@eqy Do you have any ideas why this might be happening / know anyone who could help ?

Reply:
no-pytorch repro here: https://github.com/NVIDIA/CUDALibrarySamples/issues/248